File: conformance/tests/stateful-bytestring/stateful-bytestring.runar.rs
use runar::prelude::*;

#[runar::contract]
struct MessageBoard {
    message: ByteString,
    #[readonly]
    owner: PubKey,
}

#[runar::methods(MessageBoard)]
impl MessageBoard {
    #[public]
    fn post(&mut self, new_message: ByteString) {
        self.message = new_message;
        assert!(true);
    }

    #[public]
    fn burn(&mut self, sig: Sig) {
        assert!(check_sig(sig, self.owner));
    }
}

File: conformance/tests/stateful-bytestring/stateful-bytestring.runar.go
//go:build ignore

package contract

import "runar"

type MessageBoard struct {
	runar.StatefulSmartContract
	Message runar.ByteString
	Owner   runar.PubKey `runar:"readonly"`
}

func (c *MessageBoard) Post(newMessage runar.ByteString) {
	c.Message = newMessage
	runar.Assert(true)
}

func (c *MessageBoard) Burn(sig runar.Sig) {
	runar.Assert(runar.CheckSig(sig, c.Owner))
}
